hazelcast-jet-core/src/main/java/com/hazelcast/jet/Job.java

```diff
@@ -143,17 +144,92 @@ default void join() {
     /**
      * Makes a request to cancel this job and returns. The job will complete
      * after its execution has stopped on all the nodes. If the job is
-     * already suspended, Jet will delete its runtime state so it can't be
+     * already suspended, Jet will delete its runtime state and it can't be
      * resumed.
      * <p>
      * <strong>NOTE:</strong> if the cluster becomes unstable (a member leaves or
-     * similar) while the job is in the process of being cancelled, it may end up
-     * getting restarted after the cluster has stabilized. Call {@link
+     * similar) while the job is in the process of cancellation, it may end up
+     * getting restarted after the cluster has stabilized and won't be cancelled. Call {@link
      * #getStatus()} to find out and possibly try to cancel again.
+     * <p>
+     * Job status will be {@link JobStatus#COMPLETED} after cancellation, even
+     * though the job didn't really complete. {@link Job#join()} will also
+     * return without an exception.
+     *
+     * @see #cancelAndExportState(String)
      *
      * @throws IllegalStateException if the cluster is not in a state to
      * restart the job, for example when coordinator member left and new
      * coordinator did not yet load job's metadata.
      */
     void cancel();
+
+    /**
+     * Initiates exporting of state snapshot and saves it under the given name,
+     * after which the job will be cancelled without processing any more data
+     * after the snapshot. It's similar to {@link #suspend()}, except that the
+     * job cannot be later resumed, but a new job has to be submitted with
+     * possibly modified Pipeline. To use the exported snapshot for the new
+     * job, use {@link JobConfig#setInitialSnapshotName(String)}.
+     * <p>
+     * For more information about "exported state" see {@link
+     * #exportState(String)}.
+     * <p>
+     * If the snapshot fails, the job won't be cancelled but will be suspended
+     * instead.
+     *
+     * You can call this method for suspended job: in that case the last
+     * successful snapshot will be copied and the job cancelled.
+     *
+     * Method will block until the state is fully exported, but might return
+     * before the job is fully cancelled.
+     *
+     * <p>
+     * Job status will be {@link JobStatus#COMPLETED} after cancellation, even
+     * though the job didn't really complete. However, {@link Job#join()} will
+     * throw an exception.
+
+     * TODO [viliam] finish javadoc
+     *
+     * @param name name of the snapshot
+     */
+    void cancelAndExportState(String name);
```

Comment:
is there a way to suspend and export state?

Reply:
Yes:
```
job.suspend();
while (job.getStatus() != SUSPENDED) {
    Thread.sleep(500);
}
job.exportState();
```
I don't think this use case is common to have a special support for it. When would you need it?